discussions/micro-frontends/base-paths.md

```diff
@@ -0,0 +1,150 @@
+Each of them requires configuration of build-time tooling or run-time code (or both).
+
+- Configure the build tooling to set path to assets at build-time
+- Configure the path to assets when server-side rendering
+- Use some special token for the asset path in source and then add a build step / runtime configuration to replace it
+
+### Build time base path prefixing
+
+We can configure the build tooling to prefix all asset paths in generated HTML and JavaScript.
+For example:
+
+```ts
+export default defineConfig({
+  ...
+  base: "/_fragment/todos/",
+  ...
+});
+```
+
+This is effective for ensuring that generated client side JS code is correctly prefixed; but it can miss asset paths that the build tooling is not aware of.
+The build tooling can be made aware of asset paths if the developer ensures that they are accessed via ESM imports.
+For example:
+
+```ts
+import "./App.css";
+```
+
+Paths to assets that are not known to the build tooling, such as image `src` tags, or dynamically generated import paths can be missed.
+
+This approach does not allow for the asset path to be determined at runtime based on how a fragment is being used.
+This can prevent the fragment from working both standalone and within the context of a container, since the path to assets may be different in each case.
+One solution to this is to always host the assets at the base path even when the fragment is standalone.
+
+### Dynamic base path prefixing
+
+Some frameworks, such as Qwik, allow the "base path" to be configured at runtime, when server-side rendering the HTML.
+This approach is equivalent to the build-time configuration, but gives more flexibility to change the base path depending upon usage of the fragment.
+When requesting a fragment's HTML a parameter could be added to the request specifying the asset path, which can be used when server-side rendering the HTML.
+For example:
+
+```ts
+	const url = new URL(request.url);
+	const fragmentBase = url.searchParams.get("base") ?? "/";
+	renderToStream(RootNode, {
+		...
+		envData: { request, env, fragmentBase },
+		base: fragmentBase + "build/",
+	})
+```
+
+Here the Qwik `renderToStream()` function accepts a `base` property which it uses to ensure that paths to client-side JavaScript files are correct.
+In the above example the `fragmentBase` is also added to the `envData`, which is then available via the `useEnvData()` hook in components to use when rendering asset paths in templates.
+
+### Magic base-path token
+
+If the framework does not provide a mechanism to resolve the base path at runtime, it might be possible to use a "magic token" (a unique string) for the base path to assets.
+This string will appear in the built JavaScript and the rendered HTML.
+The magic token can be replaced with the appropriate path at runtime on the server (in a Worker) when it is found in the generated HTML or JavaScript when it is served.
```

Comment:
Here at the end I think it could be quite nice to add a call to action section or something like that?

Could it make sense to create a github discussion and pointing linking to it from here (and vice versa) asking for feedback, ideas, etc...?

Reply:
Good idea. I'll add a link to Discord. I think that is the best place for informal chat, rather than RFC type discussions.